module/zfs/zio_compress.c

```diff
@@ -47,24 +47,47 @@ unsigned long zio_decompress_fail_fraction = 0;
  * Compression vectors.
  */
 zio_compress_info_t zio_compress_table[ZIO_COMPRESS_FUNCTIONS] = {
-	{"inherit",		0,	NULL,		NULL},
-	{"on",			0,	NULL,		NULL},
-	{"uncompressed",	0,	NULL,		NULL},
-	{"lzjb",		0,	lzjb_compress,	lzjb_decompress},
-	{"empty",		0,	NULL,		NULL},
-	{"gzip-1",		1,	gzip_compress,	gzip_decompress},
-	{"gzip-2",		2,	gzip_compress,	gzip_decompress},
-	{"gzip-3",		3,	gzip_compress,	gzip_decompress},
-	{"gzip-4",		4,	gzip_compress,	gzip_decompress},
-	{"gzip-5",		5,	gzip_compress,	gzip_decompress},
-	{"gzip-6",		6,	gzip_compress,	gzip_decompress},
-	{"gzip-7",		7,	gzip_compress,	gzip_decompress},
-	{"gzip-8",		8,	gzip_compress,	gzip_decompress},
-	{"gzip-9",		9,	gzip_compress,	gzip_decompress},
-	{"zle",			64,	zle_compress,	zle_decompress},
-	{"lz4",			0,	lz4_compress_zfs, lz4_decompress_zfs}
+	{"inherit",	0,	NULL,		NULL, NULL, NULL},
+	{"on",		0,	NULL,		NULL, NULL, NULL},
+	{"uncompressed", 0,	NULL,		NULL, NULL, NULL},
+	{"lzjb",	0,	lzjb_compress,	lzjb_decompress, NULL, NULL},
+	{"empty",	0,	NULL,		NULL, NULL, NULL},
+	{"gzip-1",	1,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-2",	2,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-3",	3,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-4",	4,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-5",	5,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-6",	6,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-7",	7,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-8",	8,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"gzip-9",	9,	gzip_compress,	gzip_decompress, NULL, NULL},
+	{"zle",		64,	zle_compress,	zle_decompress, NULL, NULL},
+	{"lz4",		0,	lz4_compress_zfs, lz4_decompress_zfs,
+	    NULL, NULL},
+	{"zstd",	ZIO_ZSTD_LEVEL_DEFAULT,	zstd_compress,	zstd_decompress,
+	    zstd_decompress_level, zstd_get_level},
 };
 
+uint8_t
+zio_complevel_select(spa_t *spa, enum zio_compress compress, uint8_t child,
+    uint8_t parent)
+{
+	uint8_t result;
+
+/* XXX: Allan: These need more investigation before we can assert them */
```

Comment:
Is this still the case? 
If not: remove them

It seems everything is running god damn smooth and this is NOT one of our issues during our many tests.